scripts/scil_tractogram_point_math.py

```diff
@@ -0,0 +1,182 @@
+    sft = load_tractogram_with_reference(parser, args, args.in_tractogram)
+
+    if len(sft.streamlines) == 0:
+        logging.info("Input tractogram contains no streamlines. Exiting.")
+        return
+
+    if len(args.in_dpp_name[0]) != len(args.out_name[0]):
+        parser.error('The number of in_dpp_names and out_names must be '
+                     'the same.')
+
+    data_per_point = {}
+    data_per_streamline = {}
+    for in_dpp_name, out_name in zip(args.in_dpp_name[0],
+                                     args.out_name[0]):
+        # Check to see if the data per point exists.
+        if in_dpp_name not in sft.data_per_point:
+            logging.info('Data per point {} not found in input tractogram.'
+                         .format(in_dpp_name))
+            return
+
+        # check size of first streamline data_per_point
+        if sft.data_per_point[in_dpp_name][0].shape[1] == 1:
+            is_singular = True
+        else:
+            is_singular = False
+
+        if args.operation == 'correlation' and is_singular:
+            logging.info('Correlation operation requires multivalued data per '
+                         'point. Exiting.')
+            return
+
+        # Perform the requested operation.
+        if not is_singular:
+            if args.operation == 'correlation':
+                logging.info('Performing {} across endpoint data.'.format(
+                    args.operation))
+                new_dps = perform_streamline_operation_on_endpoints(
+                    args.operation, sft, in_dpp_name)
+
+                # Adding data per streamline to new_sft
+                data_per_streamline[out_name] = new_dps
+            else:
+                # Results in new data per point
+                logging.info(
+                    'Performing {} on data from each streamine point.'.format(
+                        args.operation))
+                new_dpp = perform_streamline_operation_per_point(
+                    args.operation, sft, in_dpp_name, args.endpoints_only)
+
+                # Adding data per point to new_sft
+                data_per_point[out_name] = new_dpp
+        else:
+            # Results in new data per streamline
+            logging.info(
+                'Performing {} across each streamline.'.format(args.operation))
+            new_data_per_streamline = perform_operation_per_streamline(
+                args.operation, sft, in_dpp_name, args.endpoints_only)
+
+            # Adding data per streamline to new_sft
+            data_per_streamline[out_name] = new_data_per_streamline
+
+    new_sft = StatefulTractogram(sft.streamlines,
+                                 sft.space_attributes,
+                                 sft.space, sft.origin,
+                                 data_per_point=data_per_point,
+                                 data_per_streamline=data_per_streamline)
+
+    if len(new_sft) == 0 and not args.save_empty:
```

Comment:
You are not modifying streamlines, only the dpp and dps. So this is technically impossible.